coil-base/src/main/java/coil/fetch/FileFetcher.kt

```diff
@@ -9,9 +9,15 @@ import okio.buffer
 import okio.source
 import java.io.File
 
-internal class FileFetcher : Fetcher<File> {
+internal class FileFetcher(private val addLastModifiedToFileCacheKey: Boolean) : Fetcher<File> {
 
-    override fun key(data: File) = "${data.path}:${data.lastModified()}"
+    override fun key(data: File): String {
+        return if (addLastModifiedToFileCacheKey) {
+            "${data.path}:${data.lastModified()}"
```

Comment:
Never would have guessed this triggers a strict mode violation 😮

Reply:
Yep! It's very unlikely to cause performance issues as we're only reading the file's metadata, but technically we touch the file system from the main thread.